### ISSUE TITLE
Custom font does not work properly

### ISSUE THREAD

**Issue by itsstevenliu** [FlutterFlow/flutterflow-issues#3575]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Uploading a custom font and using it does not work as expected. I have a whole font family and none of it work. 

### Expected Behavior

Font should be per custom font set up

### Steps to Reproduce

1. Upload custom font (see sample provided file for bold font) and set it as bold
2. Create new page
3. Use custom font in text (select bold). FF IDE shows bolding
4. Local Test on iOS or TestFlight and it is very thin and does not equate to what is shown on FF IDE. Normal web 'test mode' works but does not work when testing on iOS. 

Custom font file (cannot be uploaded to GitHub as it is a .ttf file)
https://drive.google.com/file/d/1zG0ezCIZYinsSj4sRJ9Ec0hvMAMHQ0NT/view?usp=drive_link

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4ohcrq4c9JpbxK0LXyb8FvryQ7OUw8bIEvmspOEC0jF7KuBu0fdszORFFoZ9+8aVNcOVCYgkMCw9LWkYTEUu05PU2tRpRdyqlqEgzOe2i7R8nSCYe/P0Z9N9pmCBWM5reN3RYlHPRhS0IGwDmmCNy9FnLDBOu/Zwh9f6PHaOI=

### Visual documentation

![SCR-20240807-tfso](https://github.com/user-attachments/assets/43d6f304-4eb5-46f0-a99c-3d23b3b2f39e)
![IMG_2164](https://github.com/user-attachments/assets/f0e73cf4-4b10-46c6-9b62-763d0e2cb6c0)
![SCR-20240807-tdgo](https://github.com/user-attachments/assets/3116a29d-ea6c-4cec-b8ec-9a3a71990ef0)

### Environment

```markdown
- FlutterFlow version: 4.1.82
- Platform: Web and Mac
- Browser name and version: Version 127.0.6533.89 
- Operating system and version affected: 14.6 (23G80)
```

**Comment by Alezanello** [comment]
Hello!

I wasn't able to replicate the issue. I used the font you provided, uploaded as "Testing Font," as shown in the following screenshot. I ran the app in the simulator, and it loaded correctly.

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/33cf9762-8106-4d7a-a798-cce2bca727d0/37ffdad6-2bef-498a-b27c-09d596a2b953?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8zM2NmOTc2Mi04MTA2LTRkN2EtYTc5OC1jY2UyYmNhNzI3ZDAvMzdmZmRhZDYtMmJlZi00OThhLWIyN2MtMDlkNTk2YTJiOTUzIiwiaWF0IjoxNzIzMDM5MDI0LCJleHAiOjMzMjkzNTk5MDI0fQ.J94DAz8Oj4anju3gB5yegGz1lGHTa_hkiakPTc5OaCk " alt="Imagen de WhatsApp 2024-08-07 a las 10.55.10_a66773c5.jpg" width="1280" height="858" />

Best regards,<br>Azanello

**Comment by itsstevenliu** [comment]
Thanks @Alezanello 

Are you able to show us how you've set up the file on your FF? This is our view. 

<img width="1552" alt="image" src="https://github.com/user-attachments/assets/fd13a1c5-2e79-4e4a-b3a6-c032aa921def">
<img width="1552" alt="image" src="https://github.com/user-attachments/assets/234b7348-1252-4e6a-bbc5-efbb4466832c">

**Comment by Alezanello** [comment]
Of course!<br><br><br>This is the set up in Custom fonts

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/244a8094-c96a-4904-b4bb-de6955f75754/61b0f334-f8b4-4652-b675-7a826cce9c8c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8yNDRhODA5NC1jOTZhLTQ5MDQtYjRiYi1kZTY5NTVmNzU3NTQvNjFiMGYzMzQtZjhiNC00NjUyLWI2NzUtN2E4MjZjY2U5YzhjIiwiaWF0IjoxNzIzMDM5NTYyLCJleHAiOjMzMjkzNTk5NTYyfQ.OgQmTERlxKKf0WkT6mZgjLE06R3UK3i6oitPXs23rgs " alt="image.png" width="1181" height="261" />

**Comment by itsstevenliu** [comment]
Hey @Alezanello 

I think I found the source. When I upload one font file into this font family, it works. However it does not work when you upload multiple font files into the font family. Can you please try? 

You can access the font family files here: https://drive.google.com/drive/u/0/folders/1gUpDD2PqZ2H-aiLXK5_vTki6H-h2Mg7q

<img width="1203" alt="image" src="https://github.com/user-attachments/assets/d93f34e3-5c4e-4fd5-ac49-1436599975cb">

<img width="656" alt="image" src="https://github.com/user-attachments/assets/6cb52ed2-3511-4ac8-96d2-e9cde147774f">
<img width="686" alt="image" src="https://github.com/user-attachments/assets/ca7e90b7-aacf-4a1f-8e59-cfcd48973b30">

**Comment by itsstevenliu** [comment]
Not stale, waiting for reply

**Comment by Alezanello** [comment]
Hello,<br><br>I apologize for the late response. Does this issue still persist, or was it fixed in the latest FlutterFlow updates?

**Comment by itsstevenliu** [comment]
Hey @Alezanello, yes we're still experiencing the same problem.

For example, this is a screenshot straight from the app vs FlutterFlow. Per previous post, we uploaded a list of font. The example shows we assigned font file Neue Plak Text Bold to Font Weight 700. On the FF IDE, it shows the right thickness but once deployed, it is totally different and much thinner

![image](https://github.com/user-attachments/assets/7459683a-fd44-4e97-855c-0116e140270e)
![image](https://github.com/user-attachments/assets/b16846c1-ef82-41c9-a5ff-c6dbff75f3d9)
![image](https://github.com/user-attachments/assets/f6d3d2ea-bb2e-4368-80c9-0dd66b3b79d7)
